### PR TITLE
added password reset feature through otp verification

### LINKED ISSUE #286: added password reset feature through otp verification
password reset feature through otp verification

### PR DESCRIPTION
**Pull request by arpit456jain**
## Related Issue 

added password reset feature through otp verification

Closes: #286

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behaviour of your code (Write NA if not)
Any unusual behaviour of your code

## Additional Info (optional)
Any additional information you want to give

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by arpit456jain** [comment]
https://user-images.githubusercontent.com/55352601/119663748-44734c00-be50-11eb-9e5f-35891ec0fdd8.mp4

**Comment by arpit456jain** [comment]
@adarshvulli @ankitapuri i had implemented the password reset feature through otp verification .
please review this PR carefully .

File: home/views.py [hunk 1]
```diff
@@ -116,20 +121,67 @@ def error_404(request, *args, **argv):
...
 
+
+def send_email_to_user(otp,email):
+    import smtplib
+    con = smtplib.SMTP("smtp.gmail.com",587)
+    con.ehlo()
+    con.starttls()
+    admin_email = ""
+    admin_password = ""
+    con.login(admin_email,admin_password)
```
Comment (arpit456jain) [inline home/views.py:136]: Don't forgot to enter your email and password /
and make sure you enable less secure app on google account

**Comment by arpit456jain** [comment]
As soon this approved i will add some more error success msg. and will take care of some exceptions
like user can't go otp page without entering email 
like user cant go to reset password page 'with out entering otp etc etc 

Its not possible to add all these things in one PR it takes a lots of testing

**Comment by adarshvulli** [comment]
hey @arpit456jain, i see this issue during the sign up 
image rest all are fine, and if there is a possibility, we can even include the error 404 pages in some wrong routings along with the bootstrap alerts

**Comment by arpit456jain** [comment]
> hey @arpit456jain, i see this issue during the sign up
> image rest all are fine, and if there is a possibility, we can even include the error 404 pages in some wrong routings along with the bootstrap alerts

@adarshvulli don't worry about wrong routing i will handle every possible case in next PR . but i can't add 404 page since it wil work only when debug is false and if debug is false locally all css will be removed . that's why its not good to do

**Comment by arpit456jain** [comment]
and @adarshvulli i think i solved that bug in sign up will make PR for that very soon as this PR got merged

**Comment by adarshvulli** [comment]
> > hey @arpit456jain, i see this issue during the sign up
> > image rest all are fine, and if there is a possibility, we can even include the error 404 pages in some wrong routings along with the bootstrap alerts
> 
> @adarshvulli don't worry about wrong routing i will handle every possible case in next PR . but i can't add 404 page since it wil work only when debug is false and if debug is false locally all css will be removed . that's why its not good to do

okay! let me thing of this. cause at a completely wrong statement or data I see the standard error of the django but still let me even check on this!

**Comment by adarshvulli** [comment]
> and @adarshvulli i think i solved that bug in sign up will make PR for that very soon as this PR got merged

ohh okay then I will approve this!

**Review by adarshvulli — APPROVED** [review]
@ankitapuri  good to be merged

**Comment by arpit456jain** [comment]
@adarshvulli should i make PR for that bug first or complete the otp verification process

**Comment by adarshvulli** [comment]
> @adarshvulli should i make PR for that bug first or complete the otp verification process

include the correction in  the otp validation itself or comment the change in this pr

**Comment by arpit456jain** [comment]
@adarshvulli i will commit change in next PR

**Pull request merged**